src/main.rs

```diff
@@ -211,19 +228,22 @@ async fn run() -> anyhow::Result<()> {
     let file = url_path.last().context("No file found in URL")?.to_owned();
     let path = url_path[..url_path.len() - 1].join("/");
 
-    log_step(SEARCH, "Looking for checksum files...");
-
-    // Create a hashmap with the path and file to be used in the templates
-    let vars = HashMap::from([
-        ("fullpath".to_string(), url_path.join("/")),
-        ("path".to_string(), path),
-        ("file".to_string(), file.clone()),
-    ]);
-    // This shouldn't happen:
-    envsubst::validate_vars(&vars).context("unable to validate substitution variables")?;
-
-    // Create a stream of checksum downloads
-    let checksums_patterns = CHECKSUMS_FILES
+    let mut checksum = match args.checksum_source.hash_value {
+        // No hash value was passed as argument to the CLI with the --hash flag.
+        // This means we need to look for the hash in a file.
+        None => {
+            log_info("Looking for hash in checksum file");
+            // Create a hashmap with the path and file to be used in the templates
+            let vars = HashMap::from([
+                ("fullpath".to_string(), url_path.join("/")),
+                ("path".to_string(), path),
+                ("file".to_string(), file.clone()),
+            ]);
+            // This shouldn't happen:
+            envsubst::validate_vars(&vars).context("unable to validate substitution variables")?;
+
+            // Create a stream of checksum downloads
+            let checksums_patterns = CHECKSUMS_FILES
```

Comment:
The indent below is missing, please run `cargo fmt`

Reply:
Possibly extract this code into a dedicated function as it becomes quite difficult to read the code otherwise put the Some(hash) case on top for readibility.

Reply:
You're gith the indentation is strange, but this is cargo-formatted.